packages/components/organisms/f-checkout/src/store/analytics.module.js

```diff
@@ -0,0 +1,131 @@
+import Trak from '@justeat/f-trak';
+import { mapAnalyticsFieldNames } from '../services/mapper';
+
+import {
+    UPDATE_ANALYTICS_STATE,
+    UPDATE_CHANGED_FIELDS,
+    UPDATE_AUTOFILL
+} from './mutation-types';
+
+export default {
+    namespaced: true,
+
+    state: () => ({
+        serviceType: '',
+        restaurantId: '',
+        basket: {
+            id: '',
+            total: 0
+        },
+        isLoggedIn: false,
+        changedFields: [],
+        autofill: []
+    }),
+
+    actions: {
+        updateState ({ rootState, commit }, payload) {
+            const checkoutState = {
+                isLoggedIn: rootState.checkout.isLoggedIn,
+                basket: rootState.checkout.basket,
+                restaurantId: rootState.checkout.restaurantId,
+                serviceType: rootState.checkout.serviceType
+            };
+
+            if (checkoutState.isLoggedIn) {
+                commit(UPDATE_AUTOFILL, JSON.parse(JSON.stringify(payload)));
+            }
+
+            commit(UPDATE_ANALYTICS_STATE, checkoutState);
+        },
+
+        /**
+         * Pushes initial state of checkout to the dataLayer.
+         */
+        updateChangedFields ({ commit }, field) {
+            commit(UPDATE_CHANGED_FIELDS, field);
+        },
+
+        /**
+         * Pushes initial state of checkout to the dataLayer.
+         */
+        trackInitialLoad ({ state, dispatch }) {
+            window.dataLayer = window.dataLayer || [];
```

Comment:
I don't think this should go here. This, in my opinion, should be in `f-trak`.
Feel free to do it in a separate PR, but let's not forget about it.

Reply:
Good point, I'll do it after this PR if that's ok 🙂